libr/asm/arch/8051/8051_ass.c

```diff
@@ -1396,13 +1417,13 @@ int assemble_8051(RAsm *a, RAsmOp *op, char const *user_asm) {
 		free (arg[1]); arg[1] = 0; carg[1] = 0;
 		free (arg[2]); arg[2] = 0; carg[0] = 0;
 		return 0;
-	} else {
-		free (arg[0]); arg[0] = 0; carg[2] = 0;
-		free (arg[1]); arg[1] = 0; carg[1] = 0;
-		free (arg[2]); arg[2] = 0; carg[0] = 0;
-		size_t len = binp - instr;
-		r_strbuf_setbin (&op->buf, instr, len);
-
-		return binp - instr;
 	}
+
+	free (arg[0]); arg[0] = 0; carg[2] = 0;
```

Comment:
why carg is 2,1,0 ? be consistent. also those lines break the coding style as each statement should be in a separate line.

you can use `R_FREE (arg[1]);` instead of (free(arg[1]);arg[1]=0;) . also the `carg is just a local array, it doesnt needs to be deinitialized. and as long as its returning right away and all those are local vars you can just call `free` and the rest is unnecesary